modules/reactor/include/meshgenerators/TriPinHexAssemblyGenerator.h

```diff
@@ -0,0 +1,111 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "PolygonMeshGeneratorBase.h"
+
+/**
+ * This TriPinHexAssemblyGenerator object is a base class to be inherited for polygon
+ * mesh generators.
+ */
+class TriPinHexAssemblyGenerator : public PolygonMeshGeneratorBase
+{
+public:
+  static InputParameters validParams();
+
+  TriPinHexAssemblyGenerator(const InputParameters & parameters);
+
+  std::unique_ptr<MeshBase> generate() override;
+
+  enum USUAL_NUMBERS
+  {
+    UNITY = 1
+  };
+
+protected:
+  /// Radii of concentric circles in the three diamond sections
+  const std::vector<std::vector<Real>> _ring_radii;
+  /// Numbers of radial layers in each ring region in the three diamond sections
+  const std::vector<std::vector<unsigned int>> _ring_intervals;
+  /// Block ids of the ring regions in the three diamond sections
+  const std::vector<std::vector<subdomain_id_type>> _ring_block_ids;
+  /// Block names of the ring regions in the three diamond sections
+  const std::vector<std::vector<SubdomainName>> _ring_block_names;
+  /// Type of hexagon_size parameter
+  const enum class HexagonSizeStyle { apothem, radius } _hexagon_size_style;
```

Comment:
we have this enum defined a bunch of times now. 
Maybe we should move it up in PolygonMeshGeneratorBase